### PR TITLE
Remove redundant cachedBalancesController and replace cachedBalances with accountsByChainId state

### PR DESCRIPTION
**Pull request by adonesky1**
Remove redundant `cachedBalancesController` and replace `cachedBalances` with `accountsByChainId` state selectors.

The `accountsByChainId` state added to the AccountTrackerController in [this PR](https://github.com/MetaMask/metamask-extension/pull/21868) is exactly the same (modulo a slightly different shape which is easily accommodated [in the redux state selector](https://github.com/MetaMask/metamask-extension/pull/22276/files#diff-45ae84764c1d49a07813a91adc3351d6ed96bf9ee8fa2ae4c9a116d1e0cba06cR368) and we should eventually change the accountsByChainId state to actually match the cachedBalances shape since the current object with `balance` and `address` is unnecessary and redundant). Therefore we can simply remove the `cachedBalancesController` migrate its state and repoint the getCachedBalances selector to the correct new state.

### CODE REVIEW

File: .storybook/test-data.js [hunk 1]
```diff
@@ -1147,21 +1147,35 @@ const state = {
         v: '0x93',
       },
     ],
-    cachedBalances: {
+    accountsByChainId: {
       1: {
```
Comment (jiexi) [inline .storybook/test-data.js:1151]: these should probably be changed to hex chain ID? I think that's what they actually are for accountsByChainId at least

File: test/jest/mock-store.js [hunk 2]
```diff
@@ -147,9 +147,6 @@ export const createSwapsMockStore = () => {
         chainId: CHAIN_IDS.MAINNET,
         ticker: 'ETH',
       },
-      cachedBalances: {
-        [CHAIN_IDS.MAINNET]: 5,
-      },
```
Comment (jiexi) [inline test/jest/mock-store.js:152]: data so fake it doesn't even resemble the real thing at all 🙃

File: ui/components/app/alerts/unconnected-account-alert/unconnected-account-alert.test.js [hunk 3]
```diff
@@ -40,10 +40,10 @@ describe('Unconnected Account Alert', () => {
     },
   };
 
-  const cachedBalances = {
-    [CHAIN_IDS]: {
```
Comment (jiexi) [inline ui/components/app/alerts/unconnected-account-alert/unconnected-account-alert.test.js:44]: 🥲

File: ui/components/app/edit-gas-fee-popover/edit-gas-tooltip/edit-gas-tooltip.test.js [hunk 4]
```diff
@@ -34,7 +34,13 @@ const renderComponent = (componentProps) => {
   const mockStore = {
     metamask: {
       providerConfig: {},
-      cachedBalances: {},
+      accountsByChainId: {
+        '0x1': {
+          '0xAddress': {
+            balance: '0x176e5b6f173ebe66',
```
Comment (jiexi) [inline ui/components/app/edit-gas-fee-popover/edit-gas-tooltip/edit-gas-tooltip.test.js:40]:
```suggestion
            address: '0xAddress',
            balance: '0x176e5b6f173ebe66',
```

File: ui/components/app/asset-list/asset-list.test.js [hunk 5]
```diff
@@ -78,9 +78,9 @@ const render = (
           conversionRate: CONVERSION_RATE,
         },
       },
-      cachedBalances: {
+      accountsByChainId: {
         [CHAIN_IDS.MAINNET]: {
-          [selectedAddress]: balance,
+          [selectedAddress]: { balance },
```
Comment (jiexi) [inline ui/components/app/asset-list/asset-list.test.js:83]:
```suggestion
          [selectedAddress]: { address: selectedAddress, balance },
```

File: ui/components/app/asset-list/asset-list.buy-receive.test.js [hunk 6]
```diff
@@ -25,9 +25,9 @@ const render = (
     metamask: {
       ...mockState.metamask,
       providerConfig: { chainId, ticker: 'ETH' },
-      cachedBalances: {
+      accountsByChainId: {
         [CHAIN_IDS.MAINNET]: {
-          [selectedAddress]: balance,
+          [selectedAddress]: { balance },
```
Comment (jiexi) [inline ui/components/app/asset-list/asset-list.buy-receive.test.js:30]:
```suggestion
          [selectedAddress]: { address: selectedAddress, balance },
```

File: ui/components/app/alerts/unconnected-account-alert/unconnected-account-alert.test.js [hunk 7]
```diff
@@ -40,10 +40,10 @@ describe('Unconnected Account Alert', () => {
...
-      '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc': '0x0',
-      '0xec1adf982415d2ef5ec55899b9bfb8bc0f29251b': '0x0',
+  const accountsByChainId = {
+    [CHAIN_IDS.MAINNET]: {
+      '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc': { balance: '0x0' },
+      '0xec1adf982415d2ef5ec55899b9bfb8bc0f29251b': { balance: '0x0' },
```
Comment (jiexi) [inline ui/components/app/alerts/unconnected-account-alert/unconnected-account-alert.test.js:46]:
```suggestion
      '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc': { address: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc', balance: '0x0' },
      '0xec1adf982415d2ef5ec55899b9bfb8bc0f29251b': { address:'0xec1adf982415d2ef5ec55899b9bfb8bc0f29251b', balance: '0x0' },
```

File: ui/components/institutional/wrong-network-notification/wrong-network-notification.stories.js [hunk 8]
```diff
@@ -13,9 +13,9 @@ const customData = {
       type: 'test',
       chainId: toHex(3),
     },
-    cachedBalances: {
+    accountsByChainId: {
       [toHex(3)]: {
-        '0x5Ab19e7091dD208F352F8E727B6DCC6F8aBB6275': '0x0',
+        '0x5Ab19e7091dD208F352F8E727B6DCC6F8aBB6275': { balance: '0x0' },
```
Comment (jiexi) [inline ui/components/institutional/wrong-network-notification/wrong-network-notification.stories.js:18]:
```suggestion
        '0x5Ab19e7091dD208F352F8E727B6DCC6F8aBB6275': { address: '0x5Ab19e7091dD208F352F8E727B6DCC6F8aBB6275', balance: '0x0' },
```

File: ui/components/institutional/wrong-network-notification/wrong-network-notification.test.js [hunk 9]
```diff
@@ -93,9 +93,9 @@ describe('Wrong Network Notification', function () {
           type: 'test',
           chainId: toHex(3),
         },
-        cachedBalances: {
+        accountsByChainId: {
           [toHex(3)]: {
-            '0x5Ab19e7091dD208F352F8E727B6DCC6F8aBB6275': '0x0',
+            '0x5Ab19e7091dD208F352F8E727B6DCC6F8aBB6275': { balance: '0x0' },
```
Comment (jiexi) [inline ui/components/institutional/wrong-network-notification/wrong-network-notification.test.js:98]: this file needs `address` property added

File: ui/components/multichain/select-action-modal/select-action-modal.test.js [hunk 10]
```diff
@@ -62,9 +62,9 @@ describe('Select Action Modal', () => {
         chainId: CHAIN_IDS.MAINNET,
         ticker: 'ETH',
       },
-      cachedBalances: {
+      accountsByChainId: {
         '0x1': {
-          '0x1': '0x1F4',
+          '0x1': { balance: '0x1F4' },
```
Comment (jiexi) [inline ui/components/multichain/select-action-modal/select-action-modal.test.js:67]:
```suggestion
          '0x1': { address: '0x1', balance: '0x1F4' },
```

File: ui/ducks/send/send.test.js [hunk 11]
```diff
@@ -2832,9 +2832,9 @@ describe('Send Slice', () => {
               balance: '0x0',
             },
           },
-          cachedBalances: {
+          accountsByChainId: {
             [CHAIN_IDS.GOERLI]: {
-              [mockAddress1]: '0x0',
+              [mockAddress1]: { balance: '0x0' },
```
Comment (jiexi) [inline ui/ducks/send/send.test.js:2837]: same

File: ui/hooks/useAccountTotalFiatBalance.test.js [hunk 12]
```diff
@@ -52,10 +52,14 @@ const renderUseAccountTotalFiatBalance = (address) => {
...
+      accountsByChainId: {
         [CHAIN_IDS.MAINNET]: {
-          '0x0836f5ed6b62baf60706fe3adc0ff0fd1df833da': '0x041173b2c0e57d',
-          '0xd8ad671f1fcc94bcf0ebc6ec4790da35e8d5e1e1': '0x048010d1739513',
+          '0x0836f5ed6b62baf60706fe3adc0ff0fd1df833da': {
+            balance: '0x041173b2c0e57d',
+          },
+          '0xd8ad671f1fcc94bcf0ebc6ec4790da35e8d5e1e1': {
+            balance: '0x048010d1739513',
+          },
```
Comment (jiexi) [inline ui/hooks/useAccountTotalFiatBalance.test.js:62]: same

**Comment by jiexi** [comment]
small comments. otherwise looks great! woohoo cleanup 🎉

File: ui/components/institutional/wrong-network-notification/wrong-network-notification.test.js (hunk 9 above)
Comment (adonesky1) [reply ui/components/institutional/wrong-network-notification/wrong-network-notification.test.js:98]: The reason I've been leaving this out all over the place here is that state isn't actually used and should be removed as soon as your other AccountTracker PRs are merged

Comment (jiexi) [reply ui/components/institutional/wrong-network-notification/wrong-network-notification.test.js:98]: yeah true. okie dokie sounds good to me

**Review by jiexi — DISMISSED** [review]

File: .storybook/test-data.js (hunk 1 above)
Comment (adonesky1) [reply .storybook/test-data.js:1151]: agreed. Obviously this state isn't actually being used, I was debating just removing it...

Comment (adonesky1) [reply .storybook/test-data.js]: fixed here: https://github.com/MetaMask/metamask-extension/pull/22276/commits/62093184c9f8e29a07ff8c7697e0c09982b0007a

**Review by jiexi — APPROVED** [review]
🏌️ 💨 ⛳

**Review by shanejonas — APPROVED** [review]

**Pull request merged**